packages/editor/src/lib/TldrawEditor.tsx

```diff
@@ -156,13 +165,20 @@ function TldrawEditorWithOwnStore(
 		defaultName,
 	})
 
-	return <TldrawEditorWithLoadingStore {...props} store={syncedStore} />
+	return <TldrawEditorWithLoadingStore {...props} store={syncedStore} user={user} />
 }
 
 const TldrawEditorWithLoadingStore = memo(function TldrawEditorBeforeLoading({
 	store,
+	user,
 	...rest
-}: Required<TldrawEditorProps & { store: TLStoreWithStatus }, 'shapes' | 'tools'>) {
+}: Required<TldrawEditorProps & { store: TLStoreWithStatus; user: TLUser }, 'shapes' | 'tools'>) {
+	const container = useContainer()
+	if (user.userPreferences.value.isDarkMode) {
+		container.classList.remove('tl-theme__light')
+		container.classList.add('tl-theme__dark')
+	}
```

Comment:
this should probably be done in a useLayoutEffect

Reply:
taken this one